From 26913d34c91745d2f3dc89fa81eca175b286c730 Mon Sep 17 00:00:00 2001
From: Jitendra Kalsaria <jitendra.kalsaria@qlogic.com>
Date: Thu, 18 Apr 2013 19:49:53 +0000
Subject: [PATCH 633/858] qlge: Fix ethtool autoneg advertising.

Autoneg is supported on specific port types only. Fix the driver to advertise
autoneg based on the port type.

Signed-off-by: Jitendra Kalsaria <jitendra.kalsaria@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit c5e991af93e37f1b5e166f0b8a9bb99dd29e8a24)

Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>
---
 drivers/net/ethernet/qlogic/qlge/qlge_ethtool.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/qlogic/qlge/qlge_ethtool.c b/drivers/net/ethernet/qlogic/qlge/qlge_ethtool.c
index 6f316ab..0780e03 100644
--- a/drivers/net/ethernet/qlogic/qlge/qlge_ethtool.c
+++ b/drivers/net/ethernet/qlogic/qlge/qlge_ethtool.c
@@ -379,13 +379,13 @@ static int ql_get_settings(struct net_device *ndev,
 
  ecmd->supported = SUPPORTED_10000baseT_Full;
  ecmd->advertising = ADVERTISED_10000baseT_Full;
- ecmd->autoneg = AUTONEG_ENABLE;
  ecmd->transceiver = XCVR_EXTERNAL;
  if ((qdev->link_status & STS_LINK_TYPE_MASK) ==
     STS_LINK_TYPE_10GBASET) {
   ecmd->supported |= (SUPPORTED_TP | SUPPORTED_Autoneg);
   ecmd->advertising |= (ADVERTISED_TP | ADVERTISED_Autoneg);
   ecmd->port = PORT_TP;
+  ecmd->autoneg = AUTONEG_ENABLE;
  } else {
   ecmd->supported |= SUPPORTED_FIBRE;
   ecmd->advertising |= ADVERTISED_FIBRE;
-- 
1.8.3.1